src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutBase.cs

```diff
@@ -277,24 +280,46 @@ public void Hide()
 			Hide(canCancel: true);
 		}
 
-		internal void Hide(bool canCancel)
+		internal bool Hide(bool canCancel)
 		{
+			var cancel = false;
 			if (canCancel)
 			{
-				bool cancel = false;
 				OnClosing(ref cancel);
-				if (cancel)
+			}
+
+			if (!cancel && canCancel)
+			{
+				var flyout = _openFlyouts.SkipWhile(f => f != this).Skip(1).FirstOrDefault();
+				flyout?.Hide(true);
+			}
+
+			if (!cancel)
+			{
+				m_openingCanceled = true;
+
+				if (_popup != null)
 				{
-					return;
+					_popup.IsOpen = false;
 				}
-			}
+				IsOpen = false;
+
+				OnClosed();
+
+				if (_openFlyouts.IndexOf(this) == 0)
+				{
+					_openFlyouts.Remove(this);
```

Comment:
This is only called if flyout being hidden is the first one in _openFlyouts. But if it's not, we are not going to remove it from `_openFlyouts`. Why is that?

Reply:
If it isn't the first open flyout, it will wait for the flyouts "below" to close before it closes. When the "lower" flyout is closed, Hide() will be called again on this flyout and this time, the index will be 0 and it will actually close.